message/messages.go

```diff
@@ -186,69 +211,121 @@ func (mb *msgBuilder) marshal(
 	//
 	// This recursive packing allows us to avoid an extra compression on/off
 	// field in the message.
-	startTime := time.Now()
-	compressedBytes, err := mb.gzipCompressor.Compress(uncompressedMsgBytes)
-	if err != nil {
-		return nil, 0, 0, err
+	var (
+		startTime                 = time.Now()
+		compressedMsg             p2p.Message
+		opToDecompressTimeMetrics map[Op]metric.Averager
```

Comment:
We should be handling `compress`ion metrics not `decompress`ion metrics here right?